sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/LiteralsShouldNotBePassedAsLocalizedParameters.cs

```diff
@@ -0,0 +1,126 @@
+        {
+            context.RegisterSyntaxNodeActionInNonGenerated(c =>
+            {
+                var invocationSyntax = (InvocationExpressionSyntax)c.Node;
+                var methodSymbol = c.SemanticModel.GetSymbolInfo(invocationSyntax).Symbol as IMethodSymbol;
+                if (c.IsTest() ||
+                    methodSymbol == null ||
+                    invocationSyntax.ArgumentList == null)
+                {
+                    return;
+                }
+
+                if (methodSymbol.IsConsoleWrite() || methodSymbol.IsConsoleWriteLine())
+                {
+                    var firstArgument = invocationSyntax.ArgumentList.Arguments.FirstOrDefault();
+                    if (IsStringLiteral(firstArgument?.Expression, c.SemanticModel))
+                    {
+                        c.ReportDiagnostic(Diagnostic.Create(rule, firstArgument.GetLocation()));
+                    }
+                }
+                else
+                {
+                    methodSymbol.Parameters
+                        .Merge(invocationSyntax.ArgumentList.Arguments, (parameter, syntax) => new { parameter, syntax })
+                        .Where(x => x.parameter != null && x.syntax != null)
+                        .Where(x => IsLocalizable(x.parameter))
+                        .Where(x => IsStringLiteral(x.syntax.Expression, c.SemanticModel))
+                        .ToList()
+                        .ForEach(x =>
+                        {
+                            c.ReportDiagnostic(Diagnostic.Create(rule, x.syntax.GetLocation()));
+                        });
+                }
+            },
+            SyntaxKind.InvocationExpression);
+
+            context.RegisterSyntaxNodeActionInNonGenerated(c =>
+            {
+                var assignmentSyntax = (AssignmentExpressionSyntax)c.Node;
+                var propertySymbol = c.SemanticModel.GetSymbolInfo(assignmentSyntax.Left).Symbol as IPropertySymbol;
+
+                if (!c.IsTest() &&
+                    IsLocalizable(propertySymbol) &&
+                    IsStringLiteral(assignmentSyntax.Right, c.SemanticModel))
+                {
+                    c.ReportDiagnostic(Diagnostic.Create(rule, assignmentSyntax.GetLocation()));
+                }
+            },
+            SyntaxKind.SimpleAssignmentExpression);
+        }
+
+        private static bool IsStringLiteral(ExpressionSyntax expression, SemanticModel semanticModel)
+        {
+            if (expression == null)
+            {
+                return false;
+            }
+
+            var constant = semanticModel.GetConstantValue(expression);
+            return constant.HasValue && constant.Value is string;
+        }
+
+        private static bool IsLocalizable(ISymbol symbol)
+        {
+            if (symbol?.Name == null)
+            {
+                return false;
+            }
+
+            return symbol.Name.Equals("text", StringComparison.OrdinalIgnoreCase) ||

```

Comment:
Rspec says the method reports if name contains these words. Use SplitCamelCaseToWords to find the parts, or update rspec to only check the actual names.